packages/solana/src/deposit.ts

```diff
@@ -251,3 +271,54 @@ export const findVaultSwapSignature = async (
 
   return signatures[0].signature;
 };
+
+const findIdlAndInstruction = (
+  tx: ParsedTransactionWithMeta,
+): { idl: Idl; instruction: PartiallyDecodedInstruction } | { idl: null; instruction: null } => {
+  for (const inst of tx.transaction.message.instructions) {
+    switch (inst.programId.toBase58()) {
+      case mainnet.address:
+        return { idl: mainnet, instruction: inst as PartiallyDecodedInstruction };
+      case devnet.address:
+        return { idl: devnet, instruction: inst as PartiallyDecodedInstruction };
+    }
+  }
+
+  return { idl: null, instruction: null };
+};
+
+export const findVaultSwapData = async (rpcUrl: string, signature: string) => {
+  const connection = getSolanaConnection(rpcUrl);
+
+  const tx = await connection.getParsedTransaction(signature);
+
+  if (!tx) return null;
+
+  const { idl, instruction } = findIdlAndInstruction(tx);
+
+  if (instruction === null && idl === null) return null;
+
+  const coder = new BorshInstructionCoder(idl);
+  const decoded = coder.decode(instruction.data, 'base58');
+  const data = swapSchema.parse(decoded);
+
+  switch (data.name) {
+    case 'x_swap_native':
+      return {
+        sourceChain: 'Solana',
+        sourceAsset: 'Sol',
+        ...data.data,
+      };
+    case 'x_swap_token':
+      return {
+        sourceChain: 'Solana',
+        // only works because we only support the USDC. if we add more tokens
+        // we get to have more fun here
+        sourceAsset: 'SolUsdc',
+        ...data.data,
+      };
+    /* v8 ignore next 2 */
```

Comment:
🤔

Reply:
directive for the code coverage so it ignores this unreachable path that is covered by the compiler.